docs/02-app/01-building-your-application/09-authentication/index.mdx

```diff
@@ -33,84 +33,148 @@ In this section, we'll explore the process of adding basic email-password authen
 <PagesOnly>
 
 1. The user submits their credentials through a login form.
-2. The form calls a Server Action.
+2. The form sends a request that is handled by an API route.
 3. Upon successful verification, the process is completed, indicating the user's successful authentication.
 4. If verification is unsuccessful, an error message is shown.
 
 Consider a login form where users can input their credentials:
 
-```tsx filename="app/login/page.tsx" switcher
-import { authenticate } from '@/app/lib/actions'
+```tsx filename="pages/login.tsx" switcher
+import { useState } from 'react'
+import { useRouter } from 'next/router'
+
+export default function LoginPage() {
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const router = useRouter()
+
+  async function handleSubmit(e) {
+    e.preventDefault()
+    const response = await fetch('/api/auth/login', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ email, password }),
+    })
+    if (response.ok) {
+      router.push('/profile')
+    } else {
+      // Handle errors
+    }
+  }
 
-export default function Page() {
   return (
-    <form action={authenticate}>
-      <input type="email" name="email" placeholder="Email" required />
-      <input type="password" name="password" placeholder="Password" required />
+    <form onSubmit={handleSubmit}>
+      <input
+        type="email"
+        name="email"
+        placeholder="Email"
+        required
+        value={email}
+        onChange={(e) => setEmail(e.target.value)}
+      />
+      <input
+        type="password"
+        name="password"
+        placeholder="Password"
+        required
+        value={password}
```

Comment:
Rather than controlled components with state, should we do uncontrolled and read the values from the `event` in the `onSubmit`?

Reply:
cc @StephDietz

Reply:
Yes, done ✅